joomla/joomla-cms
# [6.0] Maintain usable admin on system plugin errors (Ref #39457)

Pull Request for Issue #39457.

### Summary of Changes

Joomla gracefully handles errors and presents a nice interface to the admin with the ability to "return to dashboard". However, **if the error is caused by a system plugin, the administrator interface becomes unusable**. 

Within the `ExceptionHandler` class, this pull request tries to identify a plugin directory from the error trace log, and provide a button for the admin to disable it. If the admin consents, the plugin is disabled.

The whole operation is completed within the `ExceptionHandler` class itself, because we are uncertain of whether the CMS can reach **any other existing structure**, since the plugin error is halting normal execution.

### Testing Instructions

Edit any enabled *legacy* system plugin files and write erroneous code. This pull request should be able to handle parse errors and fatal errors, but not compile errors, which are unlikely for a published plugin. Warnings and notices are not caught by the Joomla exception handler, because the administrator remains usable.

Example of a parse error:
```
print 'a but no-semi-colon'
print 'b';
```

Example of a fatal error:
```
if (JFactory::getApplication()->isAdmin()) { return true; } // Deprecated since 2014 as Brian said!
```

### Actual result BEFORE applying this Pull Request

Joomla shows the common error page and cannot recover by means of the administrator interface.

### Expected result AFTER applying this Pull Request

A message should be enqueued and shown in the message area, with a button to disable the plugin. Since that sometimes fails, depending on how deep the app execution has gone before the plugin error, another disable button has been added in the Atum administrator `error_full.php` file, at the bottom, next to "Return to Dashboard".

### Link to documentations
Please select:
